pgx/factory.go

```diff
@@ -0,0 +1,23 @@
+package pgx
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+
+	"github.com/avito-tech/go-transaction-manager/trm"
+)
+
+// NewDefaultFactory creates default trm.Transaction(pgx.Tx).
+func NewDefaultFactory(db *pgxpool.Pool) trm.TrFactory {
```

Comment:
Better use interface for Begin transaction, some people use pgx.Conn, sole pgxpool.Pool.

Reply:
I add `Transactional` interface since there is no any common interface for pgxpool.Pool and pgx.Conn.
I cannot use `Tr` (which is pgx.Tx) since this interface does not have `BeginTx` method which accepts `TxOption`